### PR TITLE
Creating page for transaction definitions in admin

### PR DESCRIPTION
**Pull request by derekdiep42**
# Basics

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [x] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [x] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

Story: https://dol-ewf.atlassian.net/browse/DSG-640

We need a page to list out all transaction definitions. This admin page has the route of /admin/transaction-definitions. 
Currently modeled after [figma](https://www.figma.com/file/PGUM4HTujZUurGrWECgBGG/Common-Components---In-Progress?node-id=59304%3A79186&mode=dev) designs of the transaction definitions page. 

Implements:
- hits new api for transaction definition changes with pagination
- implements table list for all transaction definitions
- implements table searching/filtering
- implements table sorting for key, created, and updated fields
- changes paging model to remove redundant default code

Note
- Does not implement the "+ new transaction" functionality, will be taken into account in another ticket
- Does not implement clicking a row/transaction definition, scope is out of this ticket (could be parking lot item because im not sure what screen should pop up when clicking on a row)

## Screenshot of the affected changes

Page:
Screenshot 2023-09-25 at 1 15 33 PM

Functionality:

https://github.com/Nuvalence/dsgov-web/assets/56942479/7941aaf4-2645-4173-99fb-1dec56032379

## Checklist

- [x] I have updated the documentation accordingly
- [x] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

File: libs/agency/feature/admin/src/lib/components/transaction-definitions/transaction-definitions.component.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    <nuverial-button
+      class="new-transaction-button"
+      [loading]="false"
+      (click)="false"
+      ariaLabel="Submit"
+      buttonStyle="filled"
+      buttonType="submit"
+      colorTheme="primary"
+    >
+      {{'+ New Transaction' | uppercase}}
```
Comment (davidyoung77) [inline libs/agency/feature/admin/src/lib/components/transaction-definitions/transaction-definitions.component.html:25]: Why use the uppercase pipe here, when it is hardcoded text?

File: libs/agency/feature/admin/src/lib/components/transaction-definitions/transaction-definitions.component.scss [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      @include mixins.table-styleTable();
+
+      tr {
+        cursor: pointer;
+        transition: background-color 0.3s;
+      }
+
+      th {
+        cursor: auto;
+      }
```
Comment (davidyoung77) [inline libs/agency/feature/admin/src/lib/components/transaction-definitions/transaction-definitions.component.scss:65]: Are these something that belong in the style table mixin or is this specific to this table?

File: libs/agency/feature/admin/src/lib/components/transaction-definitions/transaction-definitions.component.ts [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  public sortDirection: SortDirection = 'asc';
+  public pageSizeOptions = [10, 20, 50];
+  public pagingMetadata: IPagingMetadata | undefined;
+  public readonly pagingRequestModel: PagingRequestModel = new PagingRequestModel(
+    {
+      pageNumber: this._activatedRoute.snapshot.queryParams['pageNumber'],
+      pageSize: this._activatedRoute.snapshot.queryParams['pageSize'] || this.pageSizeOptions[0],
+      sortBy: this._activatedRoute.snapshot.queryParams['sortBy'],
+      sortOrder: this._activatedRoute.snapshot.queryParams['sortOrder'],
+    },
```
Comment (davidyoung77) [inline libs/agency/feature/admin/src/lib/components/transaction-definitions/transaction-definitions.component.ts:60]: We should be able to handle this in the model getters, I thought we were already doing this. This is a good example to clean this up in the other implementations that are using it this way

example

```
public fromSchema(paging: IPagingRequest) {
    this._sortOrder = paging.sortOrder || this._activatedRoute.snapshot.queryParams['sortOrder'] || 'ASC';
  }
public get sortOrder(): SortOrder {
    return this._sortOrder || this._activatedRoute.snapshot.queryParams['sortBy'] || 'A
  }
```

**Review by davidyoung77 — CHANGES_REQUESTED** [review]
Thanks, good work here, I have a few comments

File: libs/shared/data-access/work-api/src/lib/services/work-api-routes.service.ts [hunk 4]
```diff
@@ -107,6 +107,32 @@ export class WorkApiRoutesService extends HttpBaseService {
       })),
     );
   }
+  /**
+   * Get transaction definitions list
+   */
+  public getTransactionDefinitionsList$(
```
Comment (davidyoung77) [inline libs/shared/data-access/work-api/src/lib/services/work-api-routes.service.ts:113]: I don't see a test for this, code coverage will check for it

**Review by davidyoung77 — APPROVED** [review]
Looks like you still need to add a couple tests, but this is great

approved pending comment resolution

**Review by davidyoung77 — APPROVED** [review]

**Review by odeljanin — APPROVED** [review]
LGTM

**Pull request merged**